### ISSUE TITLE
Drupal core downloaded when not needed

### ISSUE THREAD

**Issue by mikkeschiren** [composer/installers#215]
Hi,

I pretty new to composer, so I don't know if this is a composer/installers/"or you just me doing wrong"-issue.

I have the structure then I would like to download contrib modules in the profile, and the relevant json, should be:

``` javascript
    "repositories": [
        {
          "type": "composer",
          "url": "http://drupal-packagist.webflo.io/"
        }
    ],
    "require": {
        "drupal/features": "~7",
        "drupal/wysiwyg": "~7",
        "drupal/ctools": "~7",
        "composer/installers": "~1.0"
    },
    "extra": {
      "installer-paths": {
        "modules/contrib/{$name}/": ["type:drupal-module"]
      }
    },
```

When doing this, drupal also get downloaded in the profile dir and I need to remove it. So what goes wrong here? Could anybody enlight me?

I am using

**Comment by shama** [comment]
My guess is one of those dependencies has drupal core as a dependency. Ping @RobLoach @davidbarratt

**Comment by davidbarratt** [comment]
@mikkeschiren,

This is because the module depends on drupal. This is correct because, you can't run the module without Drupal. However, the module didn't define this, @webflo's packagist does: https://github.com/webflo/packagist .

It's a little bizarre having this in Drupal 7 which doesn't really support composer, I've seen a few work-arounds, one of which is my own:
https://github.com/davidbarratt/drupal-structure
but this might be a better solution:
https://github.com/derhasi/composer-preserve-paths

regardless, Composer Installers does not support installing Drupal 7, so you'd have to use something like my Custom Installer:
https://github.com/davidbarratt/custom-installer
to specify where it should go.

I also thought about maybe subtree splitting each folder in Drupal 7 but that would be a bit of a pain.

Long story short, if you'd like to remove the Drupal dependency you'll have to open an issue here:
https://github.com/webflo/packagist/issues

**Comment by webflo** [comment]
@mikkeschiren sorry about the confusion. https://github.com/derhasi/drupal-composer-project is petty good start to use composer, drupal-packagist with drupal 7.

Feel free to open issues on https://github.com/drupal-composer/drupal-packagist

**Comment by mikkeschiren** [comment]
@davidbarratt and @webfloI think the only issue I have is that webflo's packagist adds drupal as a dependency, the rest works well, so I posted a issue on that on [webflo's packagist](https://github.com/webflo/packagist/issues/13).

And thanks for answers.

**Comment by slbmeh** [comment]
Would setting a provides on the root package satisfy the dependency, assuming you actually provide what it needs?

**Comment by thomscode** [comment]
Installing Drupal 7 via composer is now supported: https://github.com/drupal-composer/drupal-project/tree/7.x

The above link has a starting point or example you can use, that does use `composer/installers` to define the Drupal installation paths.

**Comment by alcohol** [comment]
Closing due to quiet period. Please re-open if still applicable.
